cli/internal/io.go

```diff
@@ -0,0 +1,207 @@
+package internal
+
+import (
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// CopyDir recursively copies a directory tree, attempting to preserve permissions.
+// Source directory must exist, destination directory must *not* exist.
+// Symlinks are ignored and skipped.
+func CopyDir(src string, dst string) error {
```

Comment:
We have many helpers like these ones in the beats repo for mage targets, it'd be great if we could reuse them.

Reply:
As you are more familiar with that code, could you send a follow-up PR using them?

Reply:
Addressed here: https://github.com/elastic/metricbeat-tests-poc/issues/86